pkg/network/ebpf/c/netns.h

```diff
@@ -16,36 +14,47 @@
 #define CONFIG_NET_NS
 #endif
 
-static __always_inline u32 get_netns_from_ct_net(struct net* ct_net) {
+#ifdef COMPILE_PREBUILT
+
+static __always_inline __u32 get_netns_from_sock(struct sock* sk) {
+    void* skc_net = NULL;
+    __u32 net_ns_inum = 0;
+    bpf_probe_read_kernel_with_telemetry(&skc_net, sizeof(void*), ((char*)sk) + offset_netns());
+    bpf_probe_read_kernel_with_telemetry(&net_ns_inum, sizeof(net_ns_inum), ((char*)skc_net) + offset_ino());
+    return net_ns_inum;
+}
+
+#elif defined(COMPILE_CORE) || defined(COMPILE_RUNTIME)
+
+static __always_inline u32 get_netns_ino(struct net* ns) {
     u32 net_ns_inum = 0;
-#ifdef CONFIG_NET_NS
+#if !defined(COMPILE_RUNTIME) || defined(CONFIG_NET_NS)
 #if defined(_LINUX_NS_COMMON_H) || defined(COMPILE_CORE)
-    BPF_CORE_READ_INTO(&net_ns_inum, ct_net, ns.inum);
+    BPF_CORE_READ_INTO(&net_ns_inum, ns, ns.inum);
 #else
-    BPF_CORE_READ_INTO(&net_ns_inum, ct_net, proc_inum);
+    BPF_CORE_READ_INTO(&net_ns_inum, ns, proc_inum);
 #endif
 #endif
     return net_ns_inum;
 }
 
 static __maybe_unused __always_inline u32 get_netns_from_sock(struct sock *sk) {
-    struct net *ct_net = NULL;
-#ifdef CONFIG_NET_NS
-    log_debug("get_netns_from_sock\n");
-    BPF_CORE_READ_INTO(&ct_net, sk, sk_net);
+    struct net *ns = NULL;
+#if !defined(COMPILE_RUNTIME) || defined(CONFIG_NET_NS)
```

Comment:
Does this actually get hit ? I thought we weren't using CORE with CONFIG_NET_NS ?

Reply:
Yes, this will go through for prebuilt and co-re without checking for CONFIG_NET_NS. For runtime, it will check CONFIG_NET_NS.